sync/_databricks.py

```diff
@@ -1051,7 +1053,7 @@ def _get_cluster_id_and_tasks_from_run_tasks(
         if cluster_ids:
             project_cluster_tasks = {
                 cluster_id: tasks
-                for cluster_id, tasks in all_cluster_tasks
+                for cluster_id, tasks in all_cluster_tasks.items()
```

Comment:
iterating over a dict only iterates through the keys, so this would throw an error.  The `.items()` method must be called to return key-value pairs as a tuple.